File: csharp/bird-watcher/Iteration2/BirdWatcher.cs
using System;
using System.Linq;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay) =>
        this.birdsPerDay = birdsPerDay;

    public static int[] LastWeek() => new int[] { 0, 2, 5, 3, 7, 8, 4 };

    public int Today() => birdsPerDay.Last();

    public int IncrementTodaysCount() => birdsPerDay[^1]++;

    public bool HasDayWithoutBirds() => birdsPerDay.Any(birdCount => birdCount == 0);

    public int CountForFirstDays(int numberOfDays) => birdsPerDay.Take(numberOfDays).Sum();

    public int BusyDays() => birdsPerDay.Count(birdCount => birdCount >= 5);
}

File: java/bird-watcher/Iteration2/src/main/java/BirdWatcher.java
import java.util.*; 
import java.util.stream.*;

class BirdWatcher {
    private final int[] birdsPerDay;

    public BirdWatcher(int[] birdsPerDay) {
        this.birdsPerDay = birdsPerDay.clone();
    }

    public int[] getLastWeek() {
        return new int[] { 0, 2, 5, 3, 7, 8, 4 };
    }

    public int getToday() {
        return birdsPerDay[birdsPerDay.length - 1];
    }

    public void incrementTodaysCount() {
        birdsPerDay[birdsPerDay.length - 1]++;
    }

    public boolean hasDayWithoutBirds() {
        return Arrays
            .stream(birdsPerDay)
            .anyMatch(c -> c == 0);
    }

    public int getCountForFirstDays(int numberOfDays) {
        return Arrays
            .stream(birdsPerDay)
            .limit(numberOfDays)
            .sum();
    }

    public int getBusyDays() {
        return (int)Arrays
            .stream(birdsPerDay)
            .filter(c -> c >= 5)
            .count();
    }
}
